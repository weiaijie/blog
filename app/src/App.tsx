/**
 * React Native学习测试应用主组件
 *
 * 这是应用的根组件，负责：
 * 1. 设置Redux状态管理
 * 2. 设置主题管理系统
 * 3. 初始化应用导航
 */

import React from 'react';
import { Provider } from 'react-redux'; // Redux状态管理Provider
import { store } from './store'; // Redux store配置
import { ThemeProvider } from './contexts/ThemeContext'; // 主题管理Context
import AppNavigator from './navigation/AppNavigator'; // 应用导航组件

/**
 * 应用主组件
 *
 * 组件层级结构：
 * App (根组件)
 * ├── Provider (Redux状态管理)
 * │   └── ThemeProvider (主题管理)
 * │       └── AppNavigator (导航系统)
 * │           ├── HomeScreen (首页)
 * │           ├── TodoScreen (待办事项)
 * │           ├── ProfileScreen (个人中心)
 * │           └── SettingsScreen (设置页面)
 */
const App: React.FC = () => {
  return (
    // Redux Provider: 为整个应用提供状态管理
    <Provider store={store}>
      {/* 主题Provider: 为整个应用提供主题管理功能 */}
      <ThemeProvider>
        {/* 应用导航: 管理页面间的切换和路由 */}
        <AppNavigator />
      </ThemeProvider>
    </Provider>
  );
};

export default App;
