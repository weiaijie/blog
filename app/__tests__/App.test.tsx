/**
 * App组件测试
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import App from '../src/App';

// Mock导航组件
jest.mock('../src/navigation/AppNavigator', () => {
  const React = require('react');
  const { View, Text } = require('react-native');
  
  return function MockAppNavigator() {
    return React.createElement(View, { testID: 'app-navigator' }, 
      React.createElement(Text, null, 'Mock App Navigator')
    );
  };
});

describe('App Component', () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('app-navigator')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = render(<App />);
    expect(tree).toMatchSnapshot();
  });
});
