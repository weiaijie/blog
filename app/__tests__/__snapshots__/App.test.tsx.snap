// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<View
  testID="app-navigator"
>
  <Text>
    Mock App Navigator
  </Text>
</View>
`;
